File: credit/src/main/java/ru/patterns/credit/shared/response/Account.java
package ru.patterns.credit.shared.response;

import ru.patterns.credit.shared.CreateCreditAccountResponseRaw;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.UUID;

public record Account(
        UUID id,
        LocalDateTime creationTimestamp,
        LocalDateTime blockedTimestamp,
        LocalDateTime closedTimestamp,
        UUID clientId,
        String number,
        BigDecimal balance,
        boolean isCredit
) {
}

File: core/src/main/kotlin/ru/patterns/core/domain/Account.kt
package ru.patterns.core.domain

import java.math.BigDecimal
import java.time.LocalDateTime
import java.util.UUID

/**
 * Счет клиента
 */
data class Account(
    /**
     * Идентификатор счета
     */
    val id: UUID,
    /**
     * Дата и время открытия счета
     */
    val creationTimestamp: LocalDateTime,
    /**
     * Дата и время блокировки счета. Равен 'null', если счет не заблокирован
     */
    val blockedTimestamp: LocalDateTime?,
    /**
     * Идентификатор клиента
     */
    val clientId: UUID,
    /**
     * Номер счета
     */
    val number: String,
    /**
     * Баланс счета
     */
    val balance: BigDecimal
)